### PR TITLE
Truncate long wallet names

### LINKED ISSUE #2462: Long wallet names should truncate
Let’s truncate long names with the format
`Ledger Nano X Bac…p`

image

Two instances (maybe more in other places)

Balances table view
image

Wallet switcher drawer
image

### PR DESCRIPTION
**Pull request by i-m-aditya**
Fixes #2462

### CODE REVIEW

File: packages/app-extension/src/utils/util.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const truncateName = (name: string, size: number) => {
```
Comment (tomlinton) [inline packages/app-extension/src/utils/util.ts:1]: I don't think we need to introduce a new file for this, we can add it to `@coral-xyz/common` alongside `walletAddressDisplay`. The functions are also largely the same so perhaps that could be refactored.

File: packages/chat-sdk/src/components/Attachment.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import AttachFileIcon from "@mui/icons-material/AttachFile";
 import GifIcon from "@mui/icons-material/Gif";
 import { IconButton } from "@mui/material";
 
-export const Attatchment = ({
+export const Attachment = ({
```
Comment (tomlinton) [inline packages/chat-sdk/src/components/Attachment.tsx:8]: Unrelated changes here and below

Comment (i-m-aditya) [reply packages/chat-sdk/src/components/Attachment.tsx:8]: This is because of a typo fix [PR](https://github.com/coral-xyz/backpack/pull/2353) which I pushed directly into my master, and it's not yet merged.

**Comment by aarongillett** [comment]
Instead of two full-stops `..` (if I’m interpreting the code correctly…) let’s use a single ellipsis `…`

**Comment by i-m-aditya** [comment]
> Instead of two full-stops `..` (if I’m interpreting the code correctly…) let’s use a single ellipsis `…`
I went with a single ellipsis at first, but after rebasing, I came across a function _userNameDisplay_, and it was using two dots, so I went ahead and updated walletNameDisplay too to follow the same design.

**Comment by i-m-aditya** [comment]
@hkirat Is the PR looking good?

**Comment by hkirat** [comment]
@i-m-aditya build seems to be failing and there seem to be merge conflicts could you fix those

**Comment by i-m-aditya** [comment]
> @i-m-aditya build seems to be failing and there seem to be merge conflicts could you fix those

@hkirat Was getting merge conflicts even after rebasing with the upstream master. Since the changes were small, I created a different PR for this [issue](https://github.com/coral-xyz/backpack/pull/2969).
